e2e/scripts/check-installation-state.sh

```diff
@@ -25,7 +25,49 @@ wait_for_installation() {
     done
 }
 
+wait_for_nginx_pods() {
+    ready=$(kubectl get pods -n kotsadm -o jsonpath='{.items[*].metadata.name} {.items[*].status.phase}' | grep "nginx" | grep -c Running || true)
+    counter=0
+    while [ "$ready" -lt "1" ]; do
+        if [ "$counter" -gt 36 ]; then
+            echo "nginx pods did not appear"
+            kubectl get pods -n kotsadm -o jsonpath='{.items[*].metadata.name} {.items[*].status.phase}'
+            kubectl get pods -n kotsadm
+            kubectl logs -n kotsadm -l app=kotsadm
+            return 1
+        fi
+        sleep 5
+        counter=$((counter+1))
+        echo "Waiting for nginx pods"
+        ready=$(kubectl get pods -n kotsadm -o jsonpath='{.items[*].metadata.name} {.items[*].status.phase}' | grep "nginx" | grep -c Running || true)
+        kubectl get pods -n nginx 2>&1 || true
+        echo "ready: $ready"
+    done
+}
+
+ensure_app_deployed() {
+    local version="$1"
+
+    kubectl kots get versions -n kotsadm embedded-cluster-smoke-test-staging-app
```

Comment:
you can probably use `kubectl kots get apps` and check that the status is `Ready`, but this works too i guess.

Reply:
actually, nevermind. that doesn't check a specific version.

Reply:
I'm checking the version because we upgrade to `sha-upgrade` after this, and I wanted to make sure that we're actually still pinning properly